options/options_type.h

```diff
@@ -283,6 +281,78 @@ class OptionTypeInfo {
         });
   }  // End OptionTypeInfo::Enum
 
+  // Creates an OptionTypeInfo for a Struct type.  Structs have a
+  // map of string-OptionTypeInfo associated with them that describes how
+  // to process the object for parsing, serializing, and matching.
+  // Structs also have a struct_name, which is the name of the object
+  // as registered in the parent map.
+  // When processing a struct, the option name can be specified as:
+  //   - <struct_name>       Meaning to process the entire struct.
+  //   - <struct_name.field> Meaning to process the single field
+  //   - <field>             Process the single fields
+  // The CompactionOptionsFIFO, CompactionOptionsUniversal, and LRUCacheOptions
+  // are all examples of Struct options.
+  //
+  // To create an OptionTypeInfo that is a Struct, one should:
+  // - Create a static map of string-OptionTypeInfo corresponding to the
+  //   properties of the object that can be set via the options.
+  // - Call this method passing the name and map in as parameters.
+  // Note that it is not necessary to add a new OptionType or make any
+  // other changes -- the returned object handles parsing, serialization, and
+  // comparisons.
+  //
+  // @param _offset The offset in the option object for this enum
+  // @param map The string to enum mapping for this enum
+  static OptionTypeInfo Struct(
+      const std::string& struct_name,
+      const std::unordered_map<std::string, OptionTypeInfo>* struct_map,
+      int _offset, OptionVerificationType _verification, OptionTypeFlags _flags,
+      int _mutable_offset) {
+    return OptionTypeInfo(
+        _offset, OptionType::kStruct, _verification, _flags, _mutable_offset,
```

Comment:
Usually, we do not name a variable starting with underscore. Also, for Enum, I missed the _offset in previous PR. You may change them?

Reply:
I renamed all of the member variables to end in "_" and removed any leading "_" from the parameters to the constructor